app/org/sagebionetworks/bridge/dynamodb/DynamoReportIndexDao.java

```diff
@@ -33,8 +34,13 @@ public void addIndex(ReportDataKey key) {
         DynamoReportIndex index = new DynamoReportIndex();
         index.setKey(key.getIndexKeyString());
         index.setIdentifier(key.getIdentifier());
-        
-        mapper.save(index);
+
+        // Don't recreate if it exists. There isn't a non-key attribute to use for a 
+        // conditional save operation, so load, test, then save.
+        DynamoReportIndex dbIndex = mapper.load(index);
+        if (dbIndex == null) {
+            mapper.save(index);
+        }
```

Comment:
There is a way to do a conditional save on the key. See for example DynamoUploadSchema https://github.com/Sage-Bionetworks/BridgePF/blob/develop/app/org/sagebionetworks/bridge/dynamodb/DynamoUploadSchemaDao.java#L56 Might take a little massaging to get this to work with hash-and-range key.

Upsides are that this will make the save operation atomic (as well as save a tiny amount of Read Cap). Downsides are this throws a ConditionalCheckFailedException.